### DISCUSSION TITLE
using Persist and storing File [] but getting normal object in return

### DISCUSSION THREAD

**Discussion by sushilbansal** [pmndrs/zustand#1293] [Q&A]
I am using persist middleware for my nextjs project. I selected some images (as File[]) and saved it for later (stored in zustand store). Later on when i tried to access those images i observed following issues:
1) images are returned as plain js objects
2) these File objects lose some data. The only data that is retained when i fetch it from the store: path and preview. I lose rest of the information like: name, type, size etc.

This is my store:

```
export interface FlowDataStore {
  images: File[];
  setImages: (images: File[]) => void;
}

export const useFlowDataStore = create<FlowDataStore>()(
  persist((set) => ({
    images: [],
    setImages: (images) => set({ images }),
  })),
);
```
I am not able to figure out what i am doing wrong?

**Comment by dai-shi** [comment]
Files are not serializable by default. If you want to save them in storage, you would need to provide custom serializer.

**Reply by Ethaan** [reply]
@dai-shi is there an example for this?

**Reply by dai-shi** [reply]
No, I don't remember if there's such an example. As it's not very specific to Zustand, you should be able to search something with "JavaScript File serialize JSON.stringify".

**Reply by dbritto-dev** [reply]
@sushilbansal here you go https://docs.pmnd.rs/zustand/integrations/persisting-store-data#how-do-i-use-it-with-map-and-setp

You can use same approach
